test/dialect/mysql/test_dialect.py

```diff
@@ -128,39 +128,41 @@ def test_ssl_arguments_mysqldb(self):
         from sqlalchemy.dialects.mysql import mysqldb
 
         dialect = mysqldb.dialect()
-        self._test_ssl_arguments(dialect)
+        self._test_ssl_arguments(dialect, sql_type=mysqldb)
 
     def test_ssl_arguments_oursql(self):
         from sqlalchemy.dialects.mysql import oursql
 
         dialect = oursql.dialect()
         self._test_ssl_arguments(dialect)
 
-    def _test_ssl_arguments(self, dialect):
-        kwarg = dialect.create_connect_args(
-            make_url(
-                "mysql://scott:tiger@localhost:3306/test"
-                "?ssl_ca=/ca.pem&ssl_cert=/cert.pem&ssl_key=/key.pem"
-            )
-        )[1]
+    def _test_ssl_arguments(self, dialect, sql_type="oursql"):
+        url = (
+            "mysql://scott:tiger@localhost:3306/test"
+            "?ssl_ca=/ca.pem&ssl_cert=/cert.pem&ssl_key=/key.pem"
+        )
+        expected = {
+            "passwd": "tiger",
+            "db": "test",
+            "ssl": {
+                "ca": "/ca.pem",
+                "cert": "/cert.pem",
+                "key": "/key.pem"
+            },
+            "host": "localhost",
+            "user": "scott",
+            "port": 3306
+        }
+        # add check_hostname check for mysqldb
+        if sql_type == "mysqldb":
```

Comment:
**mike bayer** (zzzeek) wrote:

this can be "if dialect.name == "pymysql"" etc

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/2774